pkg/tests/end_to_end_tests/jaeger_store_test.go

```diff
@@ -0,0 +1,41 @@
+package end_to_end_tests
+
+import (
+	"context"
+	"testing"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+	jaegertranslator "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger"
+	"github.com/stretchr/testify/require"
+
+	"github.com/timescale/promscale/pkg/jaeger/store"
+	jaegerstore "github.com/timescale/promscale/pkg/jaeger/store"
+	ingstr "github.com/timescale/promscale/pkg/pgmodel/ingestor"
+	"github.com/timescale/promscale/pkg/pgxconn"
+)
+
+// Similar to TestQueryTraces, but uses Jaeger span ingestion interface.
+func TestJaegerSpanIngestion(t *testing.T) {
+	withDB(t, "jaeger_span_store_e2e", func(db *pgxpool.Pool, t testing.TB) {
+		ingestor, err := ingstr.NewPgxIngestorForTests(pgxconn.NewPgxConn(db), nil)
+		require.NoError(t, err)
+		defer ingestor.Close()
+
+		jaegerStore := jaegerstore.New(pgxconn.NewQueryLoggingPgxConn(db), ingestor, &store.DefaultConfig)
+
+		batch, err := jaegertranslator.ProtoFromTraces(generateTestTrace())
+		require.NoError(t, err)
+		for _, b := range batch {
+			for _, s := range b.Spans {
+				// ProtoFromTraces doesn't populates span.Process because it is already been exposed by batch.Process.
+				// See https://github.com/jaegertracing/jaeger-idl/blob/05fe64e9c305526901f70ff692030b388787e388/proto/api_v2/model.proto#L152-L160
+				s.Process = b.Process
+				err = jaegerStore.SpanWriter().WriteSpan(context.Background(), s)
```

Comment:
In the context of my previous comment I'd recommend here just using `ingestor.Ingest` to ingest Jaeger spans that we transformed to traces.

Reply:
Sure, I can do that. Thats how I started, but had an afterthought of keeping jaeger specific ingest function might help incase if we want to do any optimisation in future. But I agree, it can't be designed as future proof and we can introduce an jaeger specific interface function when the need arises.

This is as simple as reverting https://github.com/timescale/promscale/pull/1543/commits/23d899c58ccfb98e0cbf90695713ae9374faa789. I will do it anyway :)

Reply:
There won't be any difference when comparing to the existing `TestQueryTraces` if we directly call IngestTrace.

The idea here is to Ingest and read jaeger span.

Reply:
Ahh right. I missed that `SpanWriter` is calling ingestor